### DISCUSSION TITLE
get_values similar to write_values

### DISCUSSION THREAD

**Discussion by raeldor** [cubewise-code/tm1py#949] [Ideas]
Hi,

We have a use-case where it would be great to be able to get multiple specific values back at a time.  Is it possible to have a 'get_values' function that is like 'write_values', but takes a list of data point tuples and returns a list similar to what is passed INTO write_values?

Thanks!

**Comment by MariusWirtz** [comment]
Sure. Feel free to open a Feature Request in the Issues section.

For now, you can use MDX with tuples but it's not as convenient.



``` python
from TM1py import TM1Service

with TM1Service(base_url="https://localhost:25734", user="admin", password="") as tm1:
    mdx = """
    SELECT
    {
    ([Period].[201912], [Country].[Austria], [Product].[Lager], [Version].[Act]),
    ([Period].[201912], [Country].[Austria], [Product].[India Pale Ale], [Version].[Act]) ,
    ([Period].[201912], [Country].[Germany], [Product].[Lager], [Version].[Act]) }
    on 0
    FROM [Sales]
    """

    df = tm1.cells.execute_mdx_dataframe(mdx)
    print(df)

```

|    | Segment      | Customer      | Sales Measure   |   Period | Country   | Product        | Version   |   Value |
|---:|:-------------|:--------------|:----------------|---------:|:----------|:---------------|:----------|--------:|
|  0 | All Segments | All Customers | Quantity        |   201912 | Austria   | Lager          | Act       |   624   |
|  1 | All Segments | All Customers | Quantity        |   201912 | Austria   | India Pale Ale | Act       |   219   |
|  2 | All Segments | All Customers | Quantity        |   201912 | Germany   | Lager          | Act       |  6451.5 |

**Reply by raeldor** [reply]
Thanks.  I actually wrote a version of get_values() based on your get_value() using this kind of format.  I'll paste my code into the feature request.
